src/rust/python-parser/lib.rs

```diff
@@ -0,0 +1,40 @@
+use std::collections::HashSet;
+
+use ruff_python_ast::{Stmt, StmtImportFrom};
+use ruff_python_parser::parse_module;
+
+#[cxx::bridge(namespace = "edgeworker::rust::python_parser")]
+mod ffi {
+
+    extern "Rust" {
+        fn get_imports(sources: &Vec<String>) -> Vec<String>;
```

Comment:
Well if I do that then the C++ name of the type is `::rust::Slice<::rust::Str const>`. I seem to be able to make one like:
```C++
  auto rust_modules = kj::heapArrayBuilder<::rust::Str const>(cpp_modules.size());
  for (auto& entry: cpp_modules) {
    rust_modules.add(entry.cStr());
  }
  ::rust::Slice<::rust::Str const> rust_slice(rust_modules.begin(), rust_modules.size());
```
I couldn't work out how to use `.as<RustCopy>` as you suggested below.

Reply:
since we're talking slices, you do not need .as<RustCopy> at all. ideally `.as<Rust>` should take care of it, but current implementation doesn't have an overload of slice of strs.

FYI `.as<T>()` is just a sugar for `T::from()`

Reply:
Sounds good, I'll try this tomorrow.

Reply:
I'm not sure how to do this because:
1. `::rust::Slice` doesn't own its data and
2. `::rust::Str` consists of different bytes than `kj::StringPtr`.

So it seems to me that isn't possible to implement an automatic converter from `kj::ArrayPtr<kj::StringPtr>` to `::rust::Slice<::rust::Str>>`. To get the `::rust::Slice` we have to make a new array and insert each `kj::StringPtr` into it, and then make a `Slice` backed by that array. The problem is that `::rust::Slice` as far as I can tell can't own anything? If I could `.attach()` the backing array to the `::rust::Slice` or somehow make it own the data, I could do it. But otherwise, someone needs to hold onto the backing data and `from()` doesn't give me a way to do that.